libs/pageserver_api/src/reltag.rs

```diff
@@ -40,21 +40,9 @@ impl PartialOrd for RelTag {
 
 impl Ord for RelTag {
     fn cmp(&self, other: &Self) -> Ordering {
-        let mut cmp = self.spcnode.cmp(&other.spcnode);
-        if cmp != Ordering::Equal {
-            return cmp;
-        }
-        cmp = self.dbnode.cmp(&other.dbnode);
-        if cmp != Ordering::Equal {
-            return cmp;
-        }
-        cmp = self.relnode.cmp(&other.relnode);
-        if cmp != Ordering::Equal {
-            return cmp;
-        }
-        cmp = self.forknum.cmp(&other.forknum);
-
-        cmp
+        // Custom ordering where we put forknum to the end of the list
+        let other_tup = (other.spcnode, other.dbnode, other.relnode, other.forknum);
+        (self.spcnode, self.dbnode, self.relnode, self.forknum).cmp(&other_tup)
```

Comment:
There's a FIXME above on line 24 for moving "forknum as last", which would allow using `derive(Ord)` for this.

Any reason why you chose this way?

Reply:
this is less invasive than implementing the FIXME. The struct has serialize and deserialize derives... maybe it is encoded in binary somewhere where the order is relevant? Of course one can find it out but that's harder :)

Reply:
Hmm yeah the Deserialize impl is not used but the Serialize impl is, and apparently it's used in the walredo binary protocol. So if we want to change this, we have to adjust it on the C side as well. Definitely more involved than this small PR.

Reply:
Could update the FIXME with these findings... Should create a test case for the reltag serialization and then reshuffle it.

Reply:
I don't think a testcase is helpful here. Pretty sure that if we break this then the walredo breaks, which can be found in the tests. I think I will just merge this after updating the FIXME.

Reply:
Yes, updating the FIXME will be enough for to keep this PR small. I was thinking the next steps is to create the test, manual serialize impl, then reshuffle the fields. Now that I looked again and realized the forknum is the first, I do it a bit surprising, but no matter, it's our code: https://github.com/neondatabase/neon/blob/a8a800af51a55a27bfad29c00a1f528e9a0ba4f2/pgxn/neon_walredo/walredoproc.c#L677-L686

Reply:
yeah that's one of the places in that file that would need updating... the comment above is also wrong as it has the forknum as the second to last, which is the *desired* state, but not the current one :laughing: